cardinal/server/shutdown.go

```diff
@@ -0,0 +1,50 @@
+package server
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"pkg.world.dev/world-engine/cardinal/ecs"
+)
+
+type ShutDownManager struct {
+	handler *Handler
+	world   *ecs.World
+}
+
+func NewShutdownManager(world *ecs.World, handler *Handler) ShutDownManager {
```

Comment:
I don't think this will work for my case here. The reason is because this manager is operating in a very specific way. It has to shutdown the server first. Then wait for that to finish, and then shut down the game loop. 

First all ongoing requests have to be processed. That could fill up the queue with Txs. Then all Txs have to be processed by the game loop. Then We shutdown. So this manager is operating on a very specific two things: Server and World.

Reply:
@jerargus forgot to address this in standup. But I'm just going to leave it for the reasoning above.